scripts/qa_crowbarsetup.sh

```diff
@@ -1574,14 +1571,8 @@ function onadmin_allocate()
         curl http://$clouddata/git/automation/scripts/qa1_nodes_reboot | bash
     fi
 
-    echo "Waiting for nodes to come up..."
-    while test $(get_all_discovered_nodes | wc -l) -lt 1 ; do
-        sleep 10
-    done
-    echo "Found one node"
-    while test $(get_all_discovered_nodes | wc -l) -lt $nodenumber ; do
-        sleep 10
-    done
+    wait_for 50 10 'test $(get_all_discovered_nodes | wc -l) -ge 1' "first node to be discovered"
+    wait_for 100 10 '[[ $(get_all_discovered_nodes | wc -l) -ge $nodenumber ]]' "all nodes to be discovered"
     local n
     for n in `get_all_discovered_nodes` ; do
```

Comment:
First wait is just for the first node, the second is for all nodes. This helped in the past a lot to see if at least one node was discovered or none.

Concerning the wording: discovered meaning `crowbar machines list | grep "^d"` lists it. So the node was seen by crowbar and no longer has an "h" as prefix.